### PR TITLE
PF TF integration tests

### PR DESCRIPTION
**Pull request by VenelinMartinov**
This moves some of the tf integration test code so that it can be used in PF and adds some PF TF integration tests.

### CODE REVIEW

File: pkg/tests/tfcheck/tfcheck.go [hunk 1]
```diff
@@ -84,6 +101,39 @@ func NewTfDriver(t pulcheck.T, dir, providerName string, prov *schema.Provider)
 	}
 }
 
+// Only exported for use in PF
+func NewTFDriverV6(t pulcheck.T, dir, providerName string, prov providerv6) *TfDriver {
```
Comment (t0yv0) [inline pkg/tests/tfcheck/tfcheck.go:105]: The signature here is a tad weird, we have

```
NewTFDriverV6
NewTFDriverSDK
```

Would it be possible instead to have

```
NewTFDriver(, <any-TF-provider>)
```

Functionality in https://github.com/hashicorp/terraform-plugin-mux can be used to mix and match and upgrade providers from SDK to protocol V6.

File: pf/tests/internal/providerbuilder/build_provider.go [hunk 2]
```diff
@@ -56,3 +62,28 @@ func (impl *Provider) Resources(ctx context.Context) []func() resource.Resource
 	}
 	return r
 }
+
+func EnsureProviderValid(prov *Provider) {
```
Comment (t0yv0) [inline pf/tests/internal/providerbuilder/build_provider.go:66]: The name and the signature here are a bit confusing. It is not running any validators from TF (what one might think based on the name). Instead it's filling out some reasonable defaults. 

Question: do we not always want to fill out reasonable defaults for `*Provider`? If so could we consider making this part of a constructor, and refactor struct `p := Provider{}` construction into `p := NewProvider()`?

**Review by t0yv0 — APPROVED** [review]
Had a few cleanup suggestions but not to stop this from merging, highly expedient to have this in, thank you!

File: pkg/tests/tfcheck/tfcheck.go (hunk 1 above)
Comment (VenelinMartinov) [reply pkg/tests/tfcheck/tfcheck.go:105]: The thinking here is that PF also supports V5, so we could potentially also want that.
These are all internal, so let's keep it as is for now and refactor as we see a need. It's not obvious that it's going to see any more use than what we currently have.

**Pull request merged**